.github/workflows/run-and-test-builds.yml

```diff
@@ -56,9 +58,23 @@ jobs:
       - name: Build example app
         run: yarn build
         working-directory: ./canary/apps/${{ matrix.path }}
-      - name: Install cypress
-        run: for n in {1..3}; do yarn install --no-lockfile && break || sleep 5; done
-        working-directory: ./canary/e2e
```

Comment:
Security note:

Using an action is more secure then using inline scripts. This is because any inputs gets passed to the action as argument, and will not be used to generate scripts.

Reply:
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-action-instead-of-an-inline-script-recommended